odk/src/main/java/org/motechproject/odk/domain/builder/FormInstanceBuilder.java

```diff
@@ -0,0 +1,160 @@
+package org.motechproject.odk.domain.builder;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+import org.joda.time.DateTime;
+import org.motechproject.odk.constant.FieldTypeConstants;
+import org.motechproject.odk.domain.FormDefinition;
+import org.motechproject.odk.domain.FormElement;
+import org.motechproject.odk.domain.FormInstance;
+import org.motechproject.odk.domain.FormValue;
+import org.motechproject.odk.domain.FormValueDateTime;
+import org.motechproject.odk.domain.FormValueDouble;
+import org.motechproject.odk.domain.FormValueGroup;
+import org.motechproject.odk.domain.FormValueInteger;
+import org.motechproject.odk.domain.FormValueString;
+import org.motechproject.odk.domain.FormValueStringList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Builder class for {@link org.motechproject.odk.domain.FormInstance}
+ */
+public class FormInstanceBuilder {
+
+    private FormDefinition formDefinition;
+    private Map<String, Object> params;
+    private String instanceId;
+
+
+    public FormInstanceBuilder(FormDefinition formDefinition, Map<String, Object> params, String instanceId) {
```

Comment:
can we javadoc this constructor, since this the main point of usage for this class. I need to know what to pass here to build a form